Tests/ApolloCacheDependentTests/FetchQueryTests.swift

```diff
@@ -241,10 +264,24 @@ class FetchQueryTests: XCTestCase {
 
         let expectation2 = self.expectation(description: "Fetching query")
 
-        client.fetch(query: query, cachePolicy: .returnCacheDataDontFetch) { (result, error) in
+        client.fetch(query: query, cachePolicy: .returnCacheDataDontFetch) { outerResult in
           defer { expectation2.fulfill() }
-          XCTAssertNil(result)
-          XCTAssertNil(error)
+          switch outerResult {
+          case .success:
+            XCTFail("This should have returned an error")
+          case .failure(let error):
+            if let resultError = error as? JSONDecodingError {
+              switch resultError {
+              case .missingValue:
+                // Correct error!
+                break
+              default:
+                XCTFail("Unexpected JSON error: \(error)")
+              }
+            } else {
+              XCTFail("Unexpected error: \(error)")
+            }
+          }
```

Comment:
This test represents a breaking change: Before if you cleared the cache and requested the same info again, you would get `nil` for both the `GraphQLResult` and the `Error`. Now you will receive an error since there is nothing to decode for that result.

Reply:
That definitely seems correct to me, I think the previous behavior was a mistake.